core/src/executor/alter/error.rs

```diff
@@ -47,4 +48,30 @@ pub enum AlterError {
 
     #[error("non-default argument should not follow the default argument")]
     NonDefaultArgumentFollowsDefaultArgument,
+
+    #[error("foreign table not found: {0}")]
+    ForeignTableNotFound(String),
+
+    #[error("foreign key column not found: {0}")]
+    ForeignKeyColumnNotFound(String),
+
+    #[error("foreign key column '{column}' of data type '{column_type:?}' does not match foreign column '{foreign_column}' of data type '{foreign_column_type:?}'")]
```

Comment:
'{column_type:?}' and '{foreign_column_type:?}' here, `ast::DataType` has Display trait impl so we don't need to use ':?'